woocommerce/woocommerce-ios
# Order Creation: Update quantity of products in the new order

Part of: #5408

## Description

During order creation, you can now update the quantity of a product in the new order. (Note that you cannot remove the product yet.)

## Changes

* In the product row (`ProductRow`/`ProductRowViewModel`):
   * Changes the `quantity` property to a `Decimal`, to match the type used in `OrderItem`.
   * Adds methods to increment and decrement the quantity, with a step value of 1 and a minimum value of 1. (A product in the order must have a quantity of at least 1; removing a product from the order is a separate action.)
   * Adds a closure `onQuantityChange` that is called when the quantity changes.
* In the new order (`NewOrderViewModel`):
   * Simplifies how we handle order items with the new representation `NewOrderItem`, which tracks the product and the editable quantity and uses those properties to create the `OrderItem` that will be sent to the remote endpoint.
   * Updates the quantity in a `NewOrderItem` when the quantity changes in its corresponding `ProductRowViewModel`.
* Adds unit tests for quantity changes in the product row and new order view models.

## Testing

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Orders tab and tap the + button. (If Simple Payments is also enabled, tap "Create order" in the bottom sheet that appears.)
4. On the New Order screen, tap the "Add product" button.
5. On the Add Product screen, tap to select a product.
6. Back on the New Order screen, tap the +/- buttons in the stepper to change the product quantity in the order. Confirm you can't change the quantity to less than 1.

## Screenshots


https://user-images.githubusercontent.com/8658164/146262489-4043f708-ad0b-463e-9b32-8a49ba13dc8f.mp4



## Submitter Checklist

Update release notes:
